### ISSUE TITLE
525 SSL Handshake Failed

### ISSUE THREAD

**Issue by therealbenpai** [badges/shields#8424]
### Are you experiencing an issue with...

shields.io

### 🐞 Description

![image](https://user-images.githubusercontent.com/69769526/191859364-3106e27f-15d3-47fb-b48a-74f41c00e495.png)

Attached above is a screenshot of me attempting to access the CDN for a badge. This is occurring quite often right now, and it lastest for over an hour at points. It occasionally works, but most of the time, I get that error message. I also posted this in the community Discord server [here](https://discordapp.com/channels/308323056592486420/477266055169835030/1022626991507263588). I don't know if this is an issue you guys can solve or not, but I'd thought I'd let you know

**Comment by chris48s** [comment]
I'm not seeing the same issue, we haven't seen any noticeable drop in traffic and I've checked all our certificates are valid.

Is anyone else seeing this problem?

**Comment by sparty182020** [comment]
I can try on my phone and see what happens

**Comment by sparty182020** [comment]
here's me accessing the "Issue Tracking" Category on the main page

![image](https://user-images.githubusercontent.com/69769526/191865189-b79f4b3e-4196-45ca-8679-cf79db52fe17.png)

**Comment by sparty182020** [comment]
It seems once one gives me that error, they all do. It starts out fine, but then it crashes. It may be a weird server issue, but it occurs for me.

**Comment by sparty182020** [comment]
If you require more information, please let me know and I'd be happy to help in anyway that I easily can

**Comment by sparty182020** [comment]
also, just to eliminate a "cache error", here's my cashe header data:
```
cache-control: private, max-age=0, no-store, no-cache, must-revalidate, post-check=0, pre-check=0
```

**Comment by sparty182020** [comment]
Ok, the issue seems to have resoved its self... so i feel that for right now, its safe to close this issue

**Comment by sparty182020** [comment]
also thanks for checking @chris48s

**Comment by chris48s** [comment]
Right.. this is back, and it is happening to me (although I don't have a _reliable_ repro). We're getting intermittent 525s - happens on some requests but not others seemingly at random. Here's the network tab for a couple of different page loads..

![Screenshot at 2022-10-24 16-15-55](https://user-images.githubusercontent.com/6025893/197565291-160f9efb-78ec-465a-8ace-3fa6dd898cdb.png)

![Screenshot at 2022-10-24 16-22-20](https://user-images.githubusercontent.com/6025893/197565323-d0fb7d78-76fe-4e17-8f7b-1d208c5f4a8d.png)

As we can see, some requests are a 200. Some are a 525 but there's no rhyme or reason to which ones. It seems to be random. This has been persistent for me for about an hour and is affecting only shields.io, not img.shields.io (for me). I've made sure nothing wierd has happened like all our instances have fallen over or something and all the certs look fine. I expect it will probably magically fix itself again in a bit.

There's a long thread at https://community.fly.io/t/cloudflare-525-error-randomly-occurs/1798 about this problem. Looks like its a known issue.

It seems like there are basically two suggested solutions:

- https://community.fly.io/t/cloudflare-525-error-randomly-occurs/1798/19
- https://community.fly.io/t/cloudflare-525-error-randomly-occurs/1798/44

Switching from strict to flexible SSL is the easiest one of those so I've tried flipping the switch for now. Doesn't seem to have immediately fixed it but I'm going to keep an eye on it. Will report back..

**Comment by chris48s** [comment]
Right. Tried setting SSL to flexible and doing an ad-hoc deploy with `force_https = false`. Left it for a bit. Hasn't made any difference. I think the next job is to try https://community.fly.io/t/cloudflare-525-error-randomly-occurs/1798/19 Not sure if I'm going to be able to slot that in tonight though.

Predictably, this problem is not affecting staging so its quite difficult to just try it out there first.

**Comment by chris48s** [comment]
Arg. Decided to go and make some dinner before getting into CloudFlare faff round 2. Came back. Everything is fixed. `200 OK`s all round, reliably.

My hunch is that switching to flexible / `force_https = false` did nothing and this just cleared up all on its own. However when dealing with an issue which is related to DNS, cache and SSL certs its tricky to know for sure.

The next thing I'm going to do is put it back how it was before (Full / `force_https = true`) and monitor things. If it breaks again in an hour or two then it was related. If it keeps working, then that did nothing.

I'll regroup after that.

**Comment by chris48s** [comment]
Everything is still working (for me). Fairly convinced flexible DNS did nothing.

I think I'm going to try and slot in working through https://community.fly.io/t/cloudflare-525-error-randomly-occurs/1798/19 anyway, but its going to be hard to tell if it has fixed the problem or not given we can't reliably reproduce the issue.

**Comment by sparty182020** [comment]
Just got on GitHub after 2 months of time off... both servers seem to be working properly for now, and because of that, I'm gonna close the issue as resolved for now and we can reopen it if needed

**Comment by chris48s** [comment]
I'm going to re-open this. I have seen this happen once more since I last posted on this issue (but not for as long) and I suspect this is affecting some subset of users on-and-off. I'd still like to try implementing https://community.fly.io/t/cloudflare-525-error-randomly-occurs/1798/19 but I didn't manage to strike while the iron was hot and it has been kind of out of sight out of mind.

If you don't want to get notifications about this issue any more you should be able to unsubscribe from it:

![Screenshot at 2022-11-26 14-07-23](https://user-images.githubusercontent.com/6025893/204092848-c3cf93fc-5e2d-4018-84c5-6e3c27bb8419.png)

If not, let me know and I'll move it to a new issue.

**Comment by chris48s** [comment]
Quick update on this, since there are probably a few new people watching the thread:

- We've been seeing more people reporting this issue over the last few days e.g:
  - https://github.com/badges/shields/issues/8884
  - https://github.com/badges/shields/issues/8888
  - https://discord.com/channels/308323056592486420/477266055169835030/1070401902497505372
- I've spent some time today trying out the steps from https://community.fly.io/t/cloudflare-525-error-randomly-occurs/1798/19 on our staging environment
- Having tried to set up shields to perform TLS termination at the container, it is clear that
  - In theory we do have [settings for it](https://github.com/badges/shields/blob/a7368b86707e4953e63d4e83ab3719e077fa724a/config/custom-environment-variables.yml#L19-L22), but they don't really work at the moment
  - Based on that, there are probably zero self-hosting users who are actively using this feature
- I have roughly got it working (in principle), but doing this for production it is going to require some tidyup and co-ordinating changes across 3 repos
- Due to other commitments it may be tricky to fit this in over the next few days

..but we are working on it.

**Comment by PyvesB** [comment]
I've not seen this happen recently, my understanding is that Chris sorted this out in linked PRs. Will go ahead and close, let us know if ever you bump into this again! 😉
